test/e2e/clusterset_test.go

```diff
@@ -330,11 +347,22 @@ var _ = ginkgo.Describe("Testing ManagedClusterSet", func() {
 				return err
 			}, eventuallyTimeout, eventuallyInterval).Should(gomega.HaveOccurred())
 		})
+
 		ginkgo.It("globalClusterSet ns should be created automatically after delete it", func() {
 			ginkgo.By("globalClusterSet ns should be created after deleted")
+
+			gomega.Eventually(func() bool {
+				_, err := kubeClient.CoreV1().Namespaces().Get(context.Background(), clustersetutils.GlobalSetNameSpace, metav1.GetOptions{})
+				if errors.IsNotFound(err) {
+					return true
+				}
+				err = kubeClient.CoreV1().Namespaces().Delete(context.Background(), clustersetutils.GlobalSetNameSpace, metav1.DeleteOptions{})
+				return errors.IsNotFound(err)
```

Comment:
when will the Delete func return a NotFound error?
Is it better to put the `err = kubeClient.CoreV1().Namespaces().Delete(context.Background(), clustersetutils.GlobalSetNameSpace, metav1.DeleteOptions{})` before the gomega.Eventually, it is not necessary to execute the delete repeatedly, right？